engine/goimage.go

```diff
@@ -347,6 +354,20 @@ func (e *GoImageEngine) Transform(img *imagefile.ImageFile, operation *Operation
 			return nil, err
 		}
 
+		if len(e.AllowedSizes) > 0 {
```

Comment:
this should be in a middleware instead of the image engine, it's only a check to perform and return a specific http code the engine image doesn't need to know.